apps/src/templates/studioHomepages/TeacherHomepage.jsx

```diff
@@ -190,6 +210,12 @@ export class UnconnectedTeacherHomepage extends Component {
     // Verify background image works for both LTR and RTL languages.
     const backgroundUrl = '/shared/images/banners/teacher-homepage-hero.jpg';
 
+    const showDonorBanner =
+      isEnglish && donorBannerName && this.state.hideTeacherOfTheYearBanner;
+
+    const showTeacherOfYearBanner =
```

Comment:
General comment, but our banner logic on the teacher dashboard is crazy. We have some [Ruby helpers for figuring out which interstitial to show](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/helpers/thank_donors_interstitial_helper.rb) (although this is also not perfect, in that we have different ones for different interstitials) -- I wonder if we could separate out our decisions about showing banners somewhere else at some point.

Reply:
Yeah the logic is confusing, I agree it would be nice to get time to clean that up.